pkg/kubelet/cm/dra/manager.go

```diff
@@ -178,10 +178,9 @@ func (m *ManagerImpl) prepareContainerResources(pod *v1.Pod, container *v1.Conta
 				return fmt.Errorf("failed to get DRA Plugin client for plugin name %s, err=%+v", driverName, err)
 			}
 
-			resourceName := fmt.Sprintf("%s-%s", pod.UID, claimName)
-			response, err := client.NodePrepareResource(context.Background(), pod.Namespace, resourceClaim.UID, resourceName, resourceClaim.Status.Allocation.ResourceHandle)
+			response, err := client.NodePrepareResource(context.Background(), pod.Namespace, resourceClaim.UID, resourceClaim.Name, resourceClaim.Status.Allocation.ResourceHandle)
 			if err != nil {
-				return fmt.Errorf("NodePrepareResource failed, pod: %s, claim UID: %s, resource: %s, resource handle: %s, err: %+v", pod.Name, resourceClaim.UID, resourceName, resourceClaim.Status.Allocation.ResourceHandle, err)
+				return fmt.Errorf("NodePrepareResource failed, pod: %s, claim UID: %s, claim name: %s, resource handle: %s, err: %+v", pod.Name, resourceClaim.UID, resourceClaim.Name, resourceClaim.Status.Allocation.ResourceHandle, err)
```

Comment:
I am not sure that we need this level of detail in the error.

Reply:
It was handy for me so far, but I'm open to suggestions.

Reply:
When writing errors, consider what the user sees. This error will show up in "kubectl describe pod/<my pod>" and probably has information that will confuse users more than it helps ("what's a resource handle?"), will have redundant information (the pod), and/or can become long (resource handle).

If you need such details during development, then check whether your logging is sufficient. That's where one should find the context for an error.